lib/prometheus_exporter/client.rb

```diff
@@ -128,6 +130,8 @@ def find_registered_metric(name, type: nil, help: nil)
     end
 
     def send_json(obj)
+      return false unless valid_naming?(obj)
```

Comment:
Should we raise instead? Otherwise the error would be pretty silent.

Also I am wondering if could be done somewhere else to avoid extra check on every `send_json`. For example when the metric is setup?

Reply:
My initial thought was to add it in `RemoteMetric` as such

```ruby
class RemoteMetric
  def valid?
     # ...
   end
end
```

But that doesn't cover the custom labels passed to the instance methods of `RemoteMetric` nor the global `custom_labels`.

Where were you thinking it would be a nice place?

About the error, I'm not sure 🤔 My initial thoughts were:
- Let's say you are using something like Sentry. You can flood your account with error events quite easily.
- Is it better to have an app crashing in the middle because of metrics?

But I'm up for discussing this.

Reply:
You are correct. I think your change is the way to go.